src/element/writer.rs

```diff
@@ -54,3 +92,30 @@ pub enum Format {
     Binary,
     // TODO a mode for Json(TextKind)
 }
+
+#[cfg(test)]
+mod tests {
+    use crate::element::owned::Element;
+    use crate::element::writer::ElementWriter;
+    use crate::ion_eq::IonEq;
+    use crate::text::text_writer::TextWriterBuilder;
+
+    use crate::{IonResult, IonWriter};
+    use nom::AsBytes;
+
+    #[test]
+    fn element_roundtrip() -> IonResult<()> {
+        let mut buffer = Vec::new();
+        let mut writer = TextWriterBuilder::new().build(&mut buffer)?;
+
+        let ion = r#"
+            null true 0 1e0 2.0 2022T foo "bar" (foo bar baz) [foo, bar, baz] {foo: true, bar: false}
+        "#;
+        let expected_elements = Element::read_all(ion.as_bytes())?;
+        writer.write_elements(&expected_elements)?;
+        writer.flush()?;
+        let actual_elements = Element::read_all(writer.output().as_bytes())?;
+        assert!(expected_elements.ion_eq(&actual_elements));
+        Ok(())
+    }
+}
```

Comment:
🗺️ This unit test used to live in `NativeElementWriter` and was moved here when that file was deleted.